perf_benchmark/benchmark_input_stream_translator_integration_test.cc

```diff
@@ -112,20 +117,60 @@ uint64_t GetNestedProtoLayer(std::string proto_msg) {
 }
 
 uint64_t GetStructProtoLayer(std::string proto_msg, std::string field_name) {
-  ::google::protobuf::Struct actual_proto;
+  pb::Struct actual_proto;
   actual_proto.ParseFromString(proto_msg);
 
   uint64_t actual_layers = 0;
-  const ::google::protobuf::Struct* it = &actual_proto;
+  const pb::Struct* it = &actual_proto;
   while (it->fields().contains(field_name)) {
     ++actual_layers;
     it = &(it->fields().at(field_name).struct_value());
   }
   return actual_layers;
 }
 
+// Parse the Grpc message binary to Json message using ResponseToJsonTranslator.
+template <class ProtoType>
+std::string ParseGrpcMessageToJsonMessage(ProtoType proto,
+                                          absl::string_view msg_type,
+                                          uint64_t num_checks, bool streaming,
+                                          uint64_t stream_size) {
+  std::string proto_binary;
+  proto.SerializeToString(&proto_binary);
+  std::string proto_binary_with_delimiter =
+      WrapGrpcMessageWithDelimiter(proto_binary);
+
+  if (streaming) {
+    // Need to make a copy otherwise the call to absl::StrAppend is undefined.
+    std::string copy = proto_binary_with_delimiter;
+    // Append stream_size - 1 proto binary to the original binary string
+    for (uint64_t i = 1; i < stream_size; ++i) {
+      absl::StrAppend(&proto_binary_with_delimiter, copy);
+    }
+  }
+  BenchmarkZeroCopyInputStream is(proto_binary_with_delimiter, num_checks);
+  // We use newline delimited to get each JSON object separated by \n instead
+  // of in an array.
+  const JsonResponseTranslateOptions options{pb::util::JsonPrintOptions(),
+                                             true};
+  ResponseToJsonTranslator translator(
+      GetBenchmarkTypeHelper().Resolver(),
+      absl::StrFormat("type.googleapis.com/%s", msg_type), streaming, &is,
+      options);
+
+  std::string message;
+  while (translator.NextMessage(&message)) {
+  }
+  EXPECT_TRUE(translator.Status().ok());
```

Comment:
Unfortunately, public googletest doens't have EXPECT_OK since it has no dependency on abseil, but there seems to be a plan for that https://github.com/abseil/abseil-cpp/issues/951.

protobuf repo only has a `GOOGLE_CHECK_OK`, but that won't work as nicely in the unit test. It will output the following message, which doesn't add information than `EXPECT_TRUE`.
```
unknown file
C++ exception with description "CHECK failed: ::google::protobuf::internal::IsOk(absl::AbortedError("Abort testing")): " thrown in the test body.
```
With `EXPECT_TRUE`
```
Value of: translator.Status().ok()
  Actual: false
Expected: true
```